crates/lib3h/src/rrdht_util/r_value.rs

```diff
@@ -0,0 +1,335 @@
+    let est_node_count = peer_record_set.peer_records.len() as f64
+        * (1.0
+            / (peer_record_set.arc_of_included_peer_records.length() as f64
+                / ARC_LENGTH_MAX as f64));
+
+    // get the estimated total coverage a.k.a r-value
+    running_avg_coverage * est_node_count
+}
+
+/// As an agent, we need to set our storage arc radius to something
+/// on the one hand, if we have enough storage and network bandwidth,
+/// we might want to set it as high as possible. On the other hand, we
+/// might want to conserve resources / power. If our network is healthy,
+/// we can scale back a bit to conserve, if it is in danger, we may want
+/// to scale up for safety. If it is anywhere in the middle, we probably
+/// want to maintain stability so we're not thrashing our resources.
+///
+/// Given current network conditions (within the sample slice given)
+/// this function will return a recommended storage arc radius.
+/// RADIUS not length. Please use `Arc::new_radius(0.into(), radius).length()`
+/// if you need to determine the length for a given radius..
+///
+/// If you provide a current_arc_radius, and the network is anywhere near
+/// stable, this function will probably return your current radius back.
+/// Otherwise it will pick a maintenance target right in the middle.
+///
+/// If the network is over replicated, this function may pick a small radius.
+/// If the network is unhealthy, or immature this function may pick a large
+/// arc radius, up to 100% in the case of a new / immature network.
+///
+/// Also, this is a bit of a naive first implementation, expect the specific
+/// algorithm / heuristics to be updated on an on-going basis.
+pub fn get_recommended_storage_arc_radius(
+    peer_record_set: &RValuePeerRecordSet,
+    target_minimum_r_value: f64,
+    target_maximum_r_value: f64,
+    current_arc_radius: Option<u32>,
+) -> u32 {
+    let pct_of_space_covered: f64 =
+        peer_record_set.arc_of_included_peer_records.length() as f64 / ARC_LENGTH_MAX as f64;
+    let interp_total_node_count =
+        peer_record_set.peer_records.len() as f64 * (1.0 / pct_of_space_covered);
+    if interp_total_node_count < target_minimum_r_value * 2.0 {
+        // consider this network immature, recommend full coverage
+        return ARC_RADIUS_MAX;
+    }
+
+    let cur_r_value = interpolate_r_value_for_given_arc(peer_record_set);
+
+    let mut count: u64 = 0;
+    let mut rad_avg: f64 = 0.0;
+
+    #[allow(clippy::explicit_counter_loop)]
+    for record in peer_record_set.peer_records.iter() {
+        count += 1;
+
+        let rad = record.storage_arc.radius();
+
+        rad_avg = (rad_avg * (count as f64 - 1.0) + f64::from(rad)) / count as f64;
+    }
+
+    let out_radius: f64 = if cur_r_value < target_minimum_r_value {
+        // our network is unhealthy! let's try to capture more!
+
+        // let's double the current average for our target mid-point
+        let new_mid_rad = rad_avg * 2.0;
```

Comment:
I sense there could be problems with this constant approach. What if we based expansion target on current arc, some factor (e.g. 2x for now, but could be based on relative uptime, available storage, bandwidth, etc. in the future), and the ratio of `cur_r_value / target_minimum_r_value` instead? If no current arc, then probably base off rad_avg (instead of current arc) as a starting point?

Reply:
Indeed, there are a billion different ways to do this. Likely most of them are better, but we need somewhere to start.

Reply:
@jmday we eventually need statistical methodologies weaved into this for initial values and how to produce estimates of R and otherwise. I think the important thing is to ensure the api is flexible enough to incorporate more advanced techniques, especially the adoption of other features to produce estimates.